### DISCUSSION TITLE
Handling values with same timestamp in Data Lake.

### DISCUSSION THREAD

**Discussion by obermeier** [apache/streampipes#1646] [Q&A] [answered]
I have some Data with the same timestamp which I want to store in the Data Lake.

Something like this:  

    {"timestamp": 1685606975, "name": "sensor1", "value": 28}
    {"timestamp": 1685606975, "name": "sensor2", "value": 30}

This leads to the problem that data in the Data Lake are replaced if they have the same timestamp.

What do you think about the idea to add the option to select a field as tag in the Data Lake config menu?
Is this to Influx specific?

This especially occurs if  Json->Array is used as format in the adapter configuraton and a  timestamp field is added.

**Comment by tenthe** [comment, marked as answer]
Hi @obermeier,
did you model the name as a `dimension property` when you defined the event schema?
If you do that, the data sink will automatically create a tag in Influx and you can have multiple events with the same timestamp.
To see the individual events, you can `group` the data in the Data Explorer.

**Reply by obermeier** [reply]
Hi @tenthe ,

thank you for the hint!
Very nice feature :)

**Reply by obermeier** [reply]
> Hi @obermeier, did you model the name as a `dimension property` when you defined the event schema? If you do that, the data sink will automatically create a tag in Influx and you can have multiple events with the same timestamp. To see the individual events, you can `group` the data in the Data Explorer.

If I understand correctly, the Schema Editor of sources provides the option to set the property scope. From a UX perspective, would it be beneficial to include this feature to the JavaScript Eval processor as well?

![image](https://github.com/apache/streampipes/assets/1632575/0a081d82-d72b-4872-95d1-19f44fe6d586)

**Reply by tenthe** [reply]
Hello @obermeier,
thank you very much for raising this issue. I think it's a valid point that we should discuss in more detail.
That's why I started a new discussion (#1654).
If you have any ideas you are welcome to write them there.

**Reply by bamf43** [reply]
Hello, I don't know if it's a bug or just my configuration, but all my dimension tags in my pipeline adapter are created as fiels instead of tag in influx.
Could someone provide some insight ?
Regards,
![image](https://github.com/user-attachments/assets/4b1c4172-1665-4cae-990a-003cc242c2af)
![image](https://github.com/user-attachments/assets/f665dfa2-0fbb-4ea0-bf28-2afc69638c68)
